### PR TITLE
Allow removal of surface net rotation

### PR DESCRIPTION
**Pull request by gassmoeller**
This builds on #3947 and adds the option to remove net surface rotation (instead of net volume rotation) to remove the nullspace. It also lets the 'rotation statistics' postprocessor compute the surface net rotation.

### CODE REVIEW

**Comment by gassmoeller** [comment]
I rebased after #3947 was merged. This should be ready for a review.

File: source/simulator/nullspace.cc [hunk 1]
```diff
@@ -348,18 +354,26 @@ namespace aspect
...
+    QGauss<dim-1> surface_quadrature(parameters.stokes_velocity_degree+1);
 
-    const unsigned int n_q_points = quadrature.size();
+    const unsigned int n_q_points = (limit_to_top_faces == false) ? quadrature.size() : surface_quadrature.size();
     UpdateFlags flags = update_quadrature_points | update_JxW_values | update_values | update_gradients;
 
-    FEValues<dim> fe (*mapping, finite_element, quadrature, flags);
+    FEValues<dim> fe_values (*mapping, finite_element, quadrature, flags);
+    FEFaceValues<dim> fe_face_values (*mapping, finite_element, surface_quadrature, flags);
+    FEValuesBase<dim> *fe((limit_to_top_faces == false)
```
Comment (tjhei) [inline source/simulator/nullspace.cc:372]: I think you can also do
```
 FEValuesBase<dim> &fe((limit_to_top_faces == false) ? dynamic_cast<FEValuesBase<dim>&>(fe_values) : dynamic_cast<FEValuesBase<dim>&>(fe_face_alues));
```
Not sure if this is better. I let you decide.

File: source/simulator/nullspace.cc [hunk 2]
```diff
@@ -379,19 +393,37 @@ namespace aspect
...
+          else
+            {
+              // We only want the output at the top boundary, so only compute it if the current cell
+              // has a face at the top boundary.
+              bool cell_at_top_boundary = false;
+              for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+                if (cell->at_boundary(f) &&
+                    (geometry_model->translate_id_to_symbol_name (cell->face(f)->boundary_id()) == "top"))
+                  {
+                    cell_at_top_boundary = true;
```
Comment (tjhei) [inline source/simulator/nullspace.cc:407]: Assert(cell_at_top_boundary==false, ExcInternalError("error, more than one top surface found in a cell."));

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
You need to update test results.

**Comment by gassmoeller** [comment]
Good point, forgot about that. Should be ready when the tester is done.

**Pull request merged**